code/datums/status_effects/xeno_buffs.dm

```diff
@@ -24,6 +27,316 @@
+
+/// Removes the status effect on death.
+/datum/status_effect/stacking/essence_link/proc/handle_death(datum/source)
+	SIGNAL_HANDLER
+	link_owner.remove_status_effect(STATUS_EFFECT_XENO_ESSENCE_LINK)
+
+/// Toggles the link depending on whether the linked xenos are still in range or not.
+/datum/status_effect/stacking/essence_link/proc/handle_dist(datum/source)
+	SIGNAL_HANDLER
+	var/within_range = get_dist(link_owner, link_target) <= link_range
+	if(within_range == was_within_range)
+		return
+	was_within_range = within_range
+	toggle_link(was_within_range)
+	link_owner.balloon_alert(link_owner, was_within_range ? ("Link reestablished") : ("Link faltering"))
+	link_target.balloon_alert(link_target, was_within_range ? ("Link reestablished") : ("Link faltering"))
+
+/// Shares the Resin Jelly buff with the linked xeno.
+/datum/status_effect/stacking/essence_link/proc/share_jelly(datum/source)
+	SIGNAL_HANDLER
+	/// Who gives the buff.
+	var/mob/living/carbon/xenomorph/buff_owner
+	/// Who receives the buff.
+	var/mob/living/carbon/xenomorph/buff_target
+
+	if(source == link_target)
+		buff_owner = link_target
+		buff_target = link_owner
+	else
+		buff_owner = link_owner
+		buff_target = link_target
+
+	buff_owner.balloon_alert(buff_owner, "Buff shared")
+	buff_target.balloon_alert(buff_target, "Buff shared")
+	buff_target.visible_message(span_notice("[buff_target]'s chitin begins to gleam with an unseemly glow..."), \
+		span_xenonotice("Through the essence link, [buff_owner] has shared their resin jelly with you."))
+	INVOKE_ASYNC(buff_target, /mob/living/carbon/xenomorph/.proc/emote, "roar")
+	buff_target.apply_status_effect(STATUS_EFFECT_RESIN_JELLY_COATING)
+
+/// Shares heals with the linked xeno.
+/datum/status_effect/stacking/essence_link/proc/share_heal(datum/source, amount, damagetype)
+	SIGNAL_HANDLER
+	/// Who gives the shared heal.
+	var/mob/living/carbon/xenomorph/heal_owner
+	/// Who receives the shared heal.
+	var/mob/living/carbon/xenomorph/heal_target
+	/// Amount of health to restore.
+	var/heal_amount
+
+	if(source == link_target)
+		heal_owner = link_target
+		heal_target = link_owner
+	else
+		heal_owner = link_owner
+		heal_target = link_target
+
+	// Prevent healing loops by temporarily removing this signal.
+	UnregisterSignal(link_owner, COMSIG_XENOMORPH_HEALED_BY_ABILITY)
+	UnregisterSignal(link_target, COMSIG_XENOMORPH_HEALED_BY_ABILITY)
+
+	heal_amount = round(clamp(amount * (shared_heal * stacks), 0, heal_target.maxHealth))
+	heal_target.visible_message(span_xenowarning("[heal_target]'s wounds are mended by faint energies."), \
+		span_xenonotice("Through the essence link, [heal_owner] has shared [heal_amount] health restoration."))
+	new /obj/effect/temp_visual/telekinesis(get_turf(heal_target))
+	heal_target.apply_healing(heal_amount)
+
+	RegisterSignal(link_owner, COMSIG_XENOMORPH_HEALED_BY_ABILITY, .proc/share_heal)
+	RegisterSignal(link_target, COMSIG_XENOMORPH_HEALED_BY_ABILITY, .proc/share_heal)
```

Comment:
This feels like a hack, I need to think on it

Reply:
Resolved.